src/main/scala/rocket/TLB.scala

```diff
@@ -79,31 +82,92 @@ class TLB(instruction: Boolean, lgMaxSize: Int, nEntries: Int)(implicit edge: TL
     val c = Bool()
   }
 
-  val totalEntries = nEntries + 1
-  val normalEntries = nEntries
-  val specialEntry = nEntries
-  val aeEntry = specialEntry - (1 << log2Floor(nEntries))
-  val valid = Reg(init = UInt(0, totalEntries))
-  val reg_entries = Reg(Vec(totalEntries, UInt(width = new Entry().getWidth)))
-  val entries = reg_entries.map(_.asTypeOf(new Entry))
+  class Entry(nSectors: Int, superpageOnly: Boolean) extends Bundle {
```

Comment:
FYI if you turn these into `val`s you shouldn't need to define your own `cloneType` anymore

Reply:
awesome